src/normalizer.ts

```diff
@@ -362,41 +307,30 @@ class V1SchemaNormalizer {
     }
 
     if (s.$ref) {
-      let ref = s.$ref
+      // we're ovewriting this string $ref with the link to the
+      // node that we find via query it may or may not have
+      // been overwritten already
       if (typeof s.$ref === 'string') {
-        ref = this.querySchemaRef(s.$ref, [...path, '$ref'])
-        if (ref) {
-          s.$ref = ref
-        }
```

Comment:
We need a lot less defensive coding. going to look around and remove checks or add more aggressive ones